### DISCUSSION TITLE
Collapse and expand sections

### DISCUSSION THREAD

**Discussion by aisbergde** [gsantner/markor#1563] [Ideas]
I tested some Android editors, including Acode and code-editor, and I was surprised about built-in syntax highlight for many languages, including Markdown and AsciiDoc. What I really like: they support line numbers and a feature to collapse and expand sections. Maybe this would be also a good idea for Markor?
![Screenshot_2022-01-15-14-12-45-925_com foxdebug acodefree](https://user-images.githubusercontent.com/9020187/149623184-e5f7bb1d-d446-4351-a5e4-a89f488010f4.jpg)

**Comment by harshad1** [comment]
One problem is that we use a plain old EditText as the underlying interface. Collapsing and expanding sections can be challenging to do well without changing to a mode powerful underlying view.

One option which _may_ work even with an EditText is this - https://developer.android.com/reference/android/text/style/DynamicDrawableSpan

**Comment by gsantner** [comment]
#922
